pom.xml

```diff
@@ -278,6 +300,80 @@
           <skipStagingRepositoryClose>true</skipStagingRepositoryClose>
         </configuration>
       </plugin>
+      <!-- 
+        Process the OSGi annotations and generate a suitable
+        MANIFEST.MF file for this bundle. This will be merged
+        with the entries from the maven-jar-plugin
+      -->
+      <plugin>
+        <groupId>biz.aQute.bnd</groupId>
+        <artifactId>bnd-maven-plugin</artifactId>
+        <version>${bnd.version}</version>
+        <executions>
+          <execution>
+            <goals>
+              <goal>bnd-process</goal>
+            </goals>
+            <configuration>
+              <bnd><![CDATA[             
+# Avoid pulling in any imports for Maven packages.
+# Their usage is isolated to two types which are
+# excluded from export. If this changes in future
+# then this Import restriction will no longer be safe
+#
+# Also mark sun.misc as an optional import. If it is
+# unavailable then the indexers will fall back to 
+# other implementations.
+
+Import-Package: \
+ !org.apache.maven.*,\
+ sun.misc;resolution:=optional,\
+ *
+          ]]></bnd>
+            </configuration>
+          </execution>
+        </executions>
+      </plugin>
+      <plugin>
+        <groupId>biz.aQute.bnd</groupId>
+        <artifactId>bnd-baseline-maven-plugin</artifactId>
+        <version>${bnd.version}</version>
+        <executions>
+          <execution>
+            <goals>
+              <goal>baseline</goal>
+            </goals>
+          </execution>
+        </executions>
+      </plugin>
+      <plugin>
+        <artifactId>maven-invoker-plugin</artifactId>
+        <version>3.2.0</version>
+          <configuration>
+            <cloneProjectsTo>${project.build.directory}/integration-test/projects</cloneProjectsTo>
+            <cloneClean>true</cloneClean>
+            <projectsDirectory>src/it/</projectsDirectory>
+            <streamLogs>true</streamLogs>
+            <localRepositoryPath>${project.build.directory}/integration-test/repo</localRepositoryPath>
+            <properties>
+              <maven.compiler.source>${maven.compiler.source}</maven.compiler.source>
+              <maven.compiler.target>${maven.compiler.target}</maven.compiler.target>
+              <org.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener>warn</org.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener>
+            </properties>
+            <goals>
+              <goal>verify</goal>
```

Comment:
The "verify" phase isn't getting executing by default on "install". Is this expected?

Reply:
So the Invoker Plugin binds automatically to:
* the `pre-integration-test` phase for the `install` goal (https://maven.apache.org/plugins/maven-invoker-plugin/install-mojo.html)
* the `integration-test` phase for the `run` goal (https://maven.apache.org/plugins/maven-invoker-plugin/run-mojo.html)

This means that the integration tests will run as part of any top-level command which triggers the `integration-test` lifecycle phase (this includes both `verify` and `install`)

The `verify` phase that you've commented on is the lifecycle target of the sub-build. This can be seen in the nested output, but it may not be obvious. 

I do see the tests run when running when I issue `mvn clean install` at the top level

Reply:
Strange, doesn't do that here. Let me look at that later.